eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/CleanUpRelationTypeEqZeroCommand.php

```diff
@@ -0,0 +1,160 @@
+use RuntimeException;
+
+class CleanUpRelationTypeEqZeroCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('ezpublish:update:legacy_storage_clean_up_relation_type_eq_zero')
+            ->setDescription('Clean up invalid relation type in Legacy Storage database')
+            ->addArgument('action', InputArgument::REQUIRED, 'Action: ' . implode(', ', $this->getAvailableActions()))
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> cleans up corrupted relations in Legacy Storage database. See: https://jira.ez.no/browse/EZP-27254
+
+<warning>During the script execution the database should not be modified.
+
+To avoid surprises you are advised to create a backup or execute a dry run:
+ 
+    %command.name% dry-run
+    
+before proceeding with actual update.</warning>
+
+Since this script can potentially run for a very long time, to avoid memory
+exhaustion run it in production environment using the <info>--env=prod</info> switch.
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $action = $input->getArgument('action');
+
+        if (!$this->isValidAction($action)) {
+            throw new RuntimeException(
+                "Action '{$action}' is not supported, use one of: " .
+                implode(', ', $this->getAvailableActions())
+            );
+        }
+
+        $totalCount = $this->getTotalCount();
+        $output->writeln('Found total relations to delete: ' . $totalCount);
+        if ($totalCount == 0) {
+            $output->writeln('Nothing to process, exiting.');
+
+            return;
+        }
+
+        if ($this->confirmExecution($input, $output)) {
+            switch ($action) {
+                case 'fix':
+                    $this->executeFix();
+                    break;
+                case 'list':
+                case 'dry-run':
+                    $this->executeList($output, $totalCount);
+                    break;
+            }
+        }
+
+        $output->writeln('');
+    }
+
+    protected function confirmExecution(InputInterface $input, OutputInterface $output)
+    {
+        $question = new ConfirmationQuestion('<question>Are you sure you want to proceed?</question> ', false);
+
+        return $this
+            ->getHelper('question')
+            ->ask($input, $output, $question);
+    }
+
+    protected function executeFix()
+    {
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $conn */
+        $conn = $this->getContainer()->get('ezpublish.connection');
+        $conn->exec('DELETE FROM ezcontentobject_link WHERE relation_type = 0');
+    }
+
+    protected function executeList(OutputInterface $output, $totalCount)
+    {
+        $output->writeln('The following relations will be removed: ');
+        $output->writeln("ID\tFCO_ID\tFCO_V\tTCO_ID\tRT");
```

Comment:
2000 relations per megabyte, then. 2 million with a gigabyte. What do you think @andrerom ? Worth it ?
Can you run it several times, @adamwojs ?